specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json

```diff
@@ -1440,6 +1452,67 @@
         }
       }
     },
+    "UpstreamAuthSettings": {
+      "description": "Upstream auth settings.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/UpstreamAuthType",
+          "description": "Gets or sets the type of auth. None or ManagedIdentity is supported now."
+        },
+        "managedIdentity": {
+          "$ref": "#/definitions/ManagedIdentitySettings",
+          "description": "Gets or sets the managed identity settings. It's required if the auth type is set to ManagedIdentity."
+        }
+      }
+    },
+    "UpstreamAuthType": {
+      "description": "Gets or sets the type of auth. None or ManagedIdentity is supported now.",
+      "enum": [
+        "None",
+        "ManagedIdentity"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "UpstreamAuthType",
+        "modelAsString": true
+      }
+    },
+    "ManagedIdentitySettings": {
+      "description": "Managed identity settings for upstream.",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The Resource indicating the App ID URI of the target resource.\r\nIt also appears in the aud (audience) claim of the issued token.",
+          "type": "string"
+        }
+      }
+    },
+    "ManagedIdentityParameters": {
+      "description": "A class represent managed identities used for request and response",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Represent the identity type: systemAssigned, userAssigned, None",
+          "type": "string"
+        },
+        "userAssignedIdentities": {
+          "description": "Get or set the user assigned identities",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/UserAssignedIdentityProperty"
+          }
+        },
+        "principalId": {
+          "description": "Get or set the principal id for the system assigned identity.\r\nOnly be used in response.",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "Get or set the tenant id for the system assigned identity.\r\nOnly be used in response",
+          "type": "string"
```

Comment:
is this readonly?

Reply:
Yes

Reply:
pls add readonly property then, same for other readyonly properties.

Reply:
`readnly` added